analyzers/src/SonarAnalyzer.CSharp/Rules/DeadStores.cs

```diff
@@ -267,11 +267,12 @@ private void ProcessVariableDeclarator(SyntaxNode instruction, HashSet<ISymbol>
                     return;
                 }
 
-                if (declarator.Initializer != null &&
-                    !IsAllowedInitialization(declarator.Initializer) &&
-                    !symbol.IsConst &&
-                    !liveOut.Contains(symbol) &&
-                    !IsUnusedLocal(symbol))
+                if (declarator.Initializer != null
+                    && !IsAllowedInitialization(declarator.Initializer)
+                    && !symbol.IsConst
+                    && symbol.RefKind() == RefKind.None
+                    && !liveOut.Contains(symbol)
```

Comment:
maybe a HashSet check is cheaper than a reflection call?

Reply:
Property read is almost as fast as reflected property read O(1). I wanted to have the symbol conditions next to each other so I'll leave it as it is.